barretenberg_wasm/src/composer.rs

```diff
@@ -149,18 +168,23 @@ impl StandardComposer {
                     &pk_ptr,
                     &cs_ptr,
                     &witness_ptr,
-                    &Value::I32(0),
+                    &Value::I32(result_ptr as i32),
                 ],
             )
             .value();
+        let proof_size: usize = proof_size.unwrap_i32() as usize;
 
-        let proof_ptr = self.barretenberg.slice_memory(0, 4);
-        let proof_ptr = u32::from_le_bytes(proof_ptr[0..4].try_into().unwrap());
+        // We then need to read the pointer at `result_ptr` to get the proof's location
+        // and then slice memory again at `proof_ptr` to get the proof data.
+        let proof_ptr = self
```

Comment:
would it be confusing to name this `proof_ptr_ptr` to denote the fact that its a pointer to a pointer, then L182 would  simply be `proof_ptr`